tests/integration/sharedHooks/useTaxon.test.js

```diff
@@ -1,34 +1,66 @@
-import { renderHook } from "@testing-library/react-native";
+import { waitFor } from "@testing-library/react-native";
+import inatjs from "inaturalistjs";
 import safeRealmWrite from "sharedHelpers/safeRealmWrite";
 import { useTaxon } from "sharedHooks";
-import factory from "tests/factory";
+import factory, { makeResponse } from "tests/factory";
 import faker from "tests/helpers/faker";
+import { renderHook } from "tests/helpers/render";
+import setupUniqueRealm from "tests/helpers/uniqueRealm";
+
+// UNIQUE REALM SETUP
+const mockRealmIdentifier = __filename;
+const { mockRealmModelsIndex, uniqueRealmBeforeAll, uniqueRealmAfterAll } = setupUniqueRealm(
+  mockRealmIdentifier
+);
+jest.mock( "realmModels/index", ( ) => mockRealmModelsIndex );
+jest.mock( "providers/contexts", ( ) => {
+  const originalModule = jest.requireActual( "providers/contexts" );
+  return {
+    __esModule: true,
+    ...originalModule,
+    RealmContext: {
+      ...originalModule.RealmContext,
+      useRealm: ( ) => global.mockRealms[mockRealmIdentifier]
+    }
+  };
+} );
+beforeAll( uniqueRealmBeforeAll );
+afterAll( uniqueRealmAfterAll );
+// /UNIQUE REALM SETUP
 
 const mockRemoteTaxon = factory( "RemoteTaxon", {
   default_photo: {
     url: faker.image.url( )
   }
 } );
 
-jest.mock( "sharedHooks/useAuthenticatedQuery", () => ( {
-  __esModule: true,
-  default: ( ) => ( {
-    data: mockRemoteTaxon
-  } )
-} ) );
-
```

Comment:
@albullington I don't think you need to review this whole PR (unless you want to), but you might want to check out the changes to this test file. Here specifically, I chose not to mock `useAuthenticatedQuery` because this is an integration test, so we *want* to see how `useTaxon` integrates with the code in this repo, like `useAuthenticatedQuery`. Instead, for most of these tests, I've mocked `inatjs` responses, because `inatjs` is outside of this repo and thus not something we need to test in an integration test.

To achieve that, I needed to make our own version of `renderHook()` that includes our `QueryClient` so React Query will actually make those requests.

I also expanded upon the tests to make sure that the API gets called when it should. The main place I struggled was testing a request *failure*, and there I had to mock React Query itself, which is not great since we're not testing our own error processing code, but I couldn't figure out another way to do it.

Reply:
Thanks for flagging. It's helpful to review this and also have it in mind as a model for future tests.